dashboard/src/components/Table/TableHeader.tsx

```diff
@@ -0,0 +1,75 @@
+import { Column } from '@tanstack/react-table';
+
+import { FormattedMessage } from 'react-intl';
+
+import { ArrowUp, ArrowDown, ArrowUpDown } from 'lucide-react';
+import { LiaQuestionCircle } from 'react-icons/lia';
+
+import { TestByCommitHash } from '@/types/tree/TreeDetails';
+import { TIndividualTest, TPathTests } from '@/types/general';
+import { formattedBreakLineValue, MessagesKey } from '@/locales/messages';
+
+import { Button } from '../ui/button';
+import { Tooltip, TooltipContent, TooltipTrigger } from '../Tooltip';
+
+interface ITableHeader {
+  column:
+    | Column<TIndividualTest>
+    | Column<TestByCommitHash>
+    | Column<TPathTests>;
+  sortable: boolean;
+  intlKey: MessagesKey;
+  intlDefaultMessage: string;
+  tooltipId?: MessagesKey;
+}
+
+export function TableHeader({
+  column,
+  sortable,
+  intlKey,
+  intlDefaultMessage,
+  tooltipId,
+}: ITableHeader): JSX.Element {
+  return (
+    <span className="flex">
+      <Button
+        variant="ghost"
+        className="justify-start px-2"
+        onClick={() => {
+          if (sortable)
+            if (column.getIsSorted() === 'asc') {
+              column.toggleSorting(true);
+            } else if (column.getIsSorted() === 'desc') {
+              column.clearSorting();
+            } else {
+              column.toggleSorting(false);
+            }
+        }}
+      >
+        <FormattedMessage
+          key={intlKey}
+          id={intlKey}
+          defaultMessage={intlDefaultMessage}
+        ></FormattedMessage>
+
+        {(column.getIsSorted() as string) === 'asc' ? (
```

Comment:
avoid type assetion, use type guard or maybe even zod to check the types instead of using as

Reply:
But I think you can probably make it type safe without runtime type check

Reply:
Done, used zod for parsing